app/src/main/java/com/cornellappdev/android/eatery/model/EateryModel.java

```diff
@@ -0,0 +1,234 @@
+package com.cornellappdev.android.eatery.model;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import com.google.android.gms.maps.model.LatLng;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.ZonedDateTime;
+
+/**
+ * Created by JC on 2/15/18. This represents a single Cafeteria (either a cafe or a dining hall)
+ */
+
+public abstract class EateryModel implements Model, Cloneable, Serializable,
+    Comparable<EateryModel> {
+
+  private boolean isHardCoded;
+  protected int id;
+  private double lat, lng;
+  protected String mName;
+  protected String mNickName;
+  private String mCloseTime;
+  protected CampusArea mArea;
+  private boolean mMatchesFilter = true;
+  private boolean mMatchesSearch = true;
+  private String mBuildingLocation;
+  protected List<String> mPayMethods;
+  private List<String> mSearchedItems;
+  private boolean openPastMidnight;
+
+  public List<String> getSearchedItems() {
+    return mSearchedItems;
+  }
+
+  public void setSearchedItems(List<String> searchedItems) {
+    this.mSearchedItems = searchedItems;
+  }
+
+  public abstract ZonedDateTime getNextOpening();
+
```

Comment:
Isn't convention to declare `abstract` methods at the top?

Reply:
Yes it is, my own codebase is a somewhat impacted by the code structure the original codebase had so I'll try to clean these sort of errors up.